doc/md/fundamentals/2-types/3-functions.md

```diff
@@ -49,110 +46,123 @@ Motoko provides different types of functions based on where in the program they
 
 Local functions run within the canister's [actor](https://internetcomputer.org/docs/motoko/fundamentals/actors-async). They cannot call other [canisters](https://internetcomputer.org/docs/building-apps/essentials/canisters). Local functions are cheap to call and execute synchronously.  
 
-```motoko no-repl
-func add(a : Nat, b : Nat) : Nat {
-    a + b
-};
+```motoko
+persistent actor CommonDivisor{
+  func gcd(a : Nat, b : Nat) : Nat {
+    var x = a;
+    var y = b;
+
+    while (y != 0) {
+      let temp = y;
+      y := x % y;
+      x := temp;
+    };
 
-let sum : Nat = add(5, 3); // Synchronous execution
+    x  // Return value is the GCD
+  };
+let greatestCommonDivisor : Nat = gcd(108, 54); // Synchronous execution
+};
 ```
 
-The type of `add` is `(Nat, Nat) -> Nat` indicating that it expects a pair of naturals  as argument and returns a natural as a result.  
+The type of `gcd` is `(Nat, Nat) -> Nat` indicating that it expects a pair of naturals as the input argument and returns a natural as a result.  
 
 **Example use case:** Local computations that do not require communication with other actors or canisters.  
 
 ## Generic functions
 
 Generic functions allow the use of type parameters, making them more flexible for using different data types.
 
-```motoko no-repl
-func identity<T>(x : T) : T = x;
+```motoko name=swap
+func swap<T, U>(t : T, u : U) : (U, T) {
+  (u, t)
+};
 
-let num = identity<Nat>(42);  
-let text = identity<Text>("Hello");  
+swap<Nat, Text>(42, "ICP"); // ("ICP", 42)
 ```
 
-The type of identity is `<T>T -> T`, indicating that when supplied with any type `T`, and argument of type `T`, it returns a result of the same type `T`.
+The type of `swap` is `<T, U> (T, U) -> (U, T)`, indicating it can accept any two types and return them in reversed order.
```

Comment:
```suggestion
The type of `swap` is `<T, U> (T, U) -> (U, T)`, indicating it can accept any  two types `T` and `U`, a  value of tuple type `(T, U)` and return a tuple of type `(U, T)`, with the elements swapped.
```